src/Libraries/Microsoft.Extensions.AI.AzureAIInference/AzureAIInferenceChatClient.cs

```diff
@@ -441,7 +443,9 @@ private IEnumerable<ChatRequestMessage> ToAzureAIInferenceChatMessages(IEnumerab
             }
             else if (input.Role == ChatRole.User)
             {
-                yield return new ChatRequestUserMessage(GetContentParts(input.Contents));
+                yield return input.Contents.All(c => c is TextContent) ?
+                    new ChatRequestUserMessage(string.Concat(input.Contents)) :
```

Comment:
I know it's a super-niche case, but should we treat separate `Contents` entries as if they have some kind of separator between them, for example a linebreak?

Concatenating with no separator at all could theoretically result in different meanings than were intended.

Reply:
Do we have evidence to suggest that's desirable? With streaming, for example, each TextContent that's produced contains its own separators where necessary, e.g. the streamed contents for "hello world" will often be an update for "hello " and an update for "world", which we then represent as a TextContent for "hello " and a TextContent for "world". I wouldn't want to then artificially insert additional whitespace.

Reply:
Another option is we could just pass through the bug/limitation from AzureAIInferenceChatClient without changing any semantics:

 * If the supplied value is exactly one TextContent, we could use the `string` overload that works universally
 * Otherwise, we use the `IEnumerable<ChatMessageContentItem>` overload that may or may not work depending on the backend model. Later this might get fixed in `AzureAIInferenceChatClient` or in the service itself to work universally as long as all the content is text, but that happens independently from this library.

I know in practice it likely isn't harmful to polyfill over the bug/limitation (and if it becomes fixed in the underlying client/service in the future, we could remove our polyfill without thinking of that as a breaking change), so I don't feel strongly about this.

Reply:
Ok. Until we get any feedback to the contrary, I'm going to leave it as-is.